pkg/imageregistry/imageregistry.go

```diff
@@ -0,0 +1,328 @@
+var imageRegistryNamespace string = "openshift-image-registry"
+var imageRegistryCoName string = "image-registry"
+
+// Builder provides a struct for imageRegistry object from the cluster and a imageRegistry definition.
+type Builder struct {
+	// imageRegistry definition, used to create the imageRegistry object.
+	Definition *imageregistryv1.Config
+	// Created imageRegistry object.
+	Object *imageregistryv1.Config
+	// api client to interact with the cluster.
+	apiClient *clients.Settings
+}
+
+// Pull retrieves an existing imageRegistry object from the cluster.
+func Pull(apiClient *clients.Settings) (*Builder, error) {
+	glog.V(100).Infof(
+		"Pulling imageRegistry object name: %s", imageRegistryObjName)
+
+	builder := Builder{
+		apiClient: apiClient,
+		Definition: &imageregistryv1.Config{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: imageRegistryObjName,
+			},
+		},
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("imageRegistry object %s doesn't exist", imageRegistryObjName)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// Get fetches existing imageRegistry from cluster.
+func (builder *Builder) Get() (*imageregistryv1.Config, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Pulling existing imageRegistry with name %s from cluster", builder.Definition.Name)
+
+	imageRegistry := &imageregistryv1.Config{}
+	err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{
+		Name: builder.Definition.Name,
+	}, imageRegistry)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return imageRegistry, nil
+}
+
+// Exists checks whether the given imageRegistry exists.
+func (builder *Builder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if imageRegistry %s exists", builder.Definition.Name)
+
+	var err error
+	builder.Object, err = builder.Get()
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// Update renovates a imageRegistry in the cluster and stores the created object in struct.
+func (builder *Builder) Update() (*Builder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Updating the imageRegistry %s", builder.Definition.Name)
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("imageRegistry object %s doesn't exist", builder.Definition.Name)
+	}
+
+	err := builder.apiClient.Update(context.TODO(), builder.Definition)
+	if err == nil {
+		builder.Object = builder.Definition
+	}
+
+	return builder, err
+}
+
+// GetManagementState fetches imageRegistry ManagementState.
+func (builder *Builder) GetManagementState() (v1.ManagementState, error) {
+	glog.V(100).Infof("Getting imageRegistry ManagementState configuration")
+
+	if !builder.Exists() {
+		return "", fmt.Errorf("imageRegistry object doesn't exist")
+	}
+
+	imageRegistryManagementState := builder.Object.Spec.ManagementState
+
+	return imageRegistryManagementState, nil
```

Comment:
No need for imageRegistryManagementState.
You could just:
return builder.Object.Spec.ManagementState, nil

Reply:
np, I just believe that you know that it is a bad practice

Reply:
I agree it's minor comment  but from my perspective we don't need a variable that is not in use.